src/Cm/RedisSession/Handler.php

```diff
@@ -366,13 +383,38 @@ public function __construct(ConfigInterface $config, LoggerInterface $logger, $r
             }
         }
         else {
-            $this->_redis = new \Credis_Client($host, $port, $timeout, $persistent, 0, $pass);
+            if (($config instanceof ClusterConfigInterface) && ($config->isCluster())) {
+                $this->_redis = new \Credis_ClusterClient(
+                    $config->getClusterName(),
+                    $config->getClusterSeeds(),
+                    $timeout,
+                    0,
+                    $config->getClusterUsePersistentConnection(),
+                    $pass,
+                    $username,
```

Comment:
Username is optional

Reply:
Yes, and it is currently null if config is not UsernameConfigInterface.
See here.
https://github.com/JacobBrownAustin/php-redis-session-abstract/pull/1/files#diff-fef2780a2c99c6526c10142058e117d6b01098444f4ddcee6451b4d4ca23f8f5R294

Previously, it was just using the default parameter value of null, so in case the config is not UsernameConfigInterface, then behaviour is same as previous.

Reply:
Here's where you can see the default parameter value of null.
https://github.com/JacobBrownAustin/credis/pull/1/files#diff-378337b67f4e73113804810a9ce436cb8e1ee50fa9631cc79352d03be87fe8a0R61

Reply:
In this case, which is my new code, it is being passed to RedisCluster, 
['user' => $this->authUsername, 'pass' => $this->authUsername],
But in the extension, it only uses it if the length is non-zero.
https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L132